charts/flyte-core/templates/agent/rbac.yaml

```diff
@@ -0,0 +1,17 @@
+{{- if .Values.flyteagent.enabled }}
+---
+{{- if .Values.flyteagent.serviceAccount.create }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "flyteagent.name" . }}
+  namespace: {{ template "flyte.namespace" . }}
+  labels: {{ include "flyteagent.labels" . | nindent 4 }}
+  {{- with .Values.flyteagent.serviceAccount.annotations }}
+  annotations: {{ tpl (toYaml .) $ | nindent 4 }}
+  {{- end}}
+{{- with .Values.flyteagent.serviceAccount.imagePullSecrets }}
+imagePullSecrets: {{ tpl (toYaml .) $ | nindent 2 }}
```

Comment:
Shouldn't this be under spec

Reply:
No, https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#add-image-pull-secret-to-service-account. We add it to the spec if it's a pod